albumentations/augmentations/transforms.py

```diff
@@ -3320,3 +3321,61 @@ def get_transform_init_args_names(self):
             "mask_fill_value",
             "random_offset",
         )
+
+
+class AugMix(ImageOnlyTransform):
+    """Augmentations mix to Improve Robustness and Uncertainty.
+    Args:
+        image (np.ndarray): Raw input image of shape (h, w, c)
+        severity (int): Severity of underlying augmentation operators.
+        width (int): Width of augmentation chain
+        depth (int): Depth of augmentation chain. -1 enables stochastic depth uniformly
+          from [1, 3]
+        alpha (float): Probability coefficient for Beta and Dirichlet distributions.
+        augmentations (list of augmentations): Augmentations that need to mix and perform.
+    Targets:
+        image
+    Image types:
+        uint8, float32
+
+    Reference:
+    |  https://arxiv.org/abs/1912.02781
+    |  https://github.com/google-research/augmix
+    """
+
+    def __init__(self, width=2, depth=2, alpha=0.5, augmentations=[HorizontalFlip()], always_apply=False, p=0.5):
+        super(AugMix, self).__init__(always_apply, p)
+        self.width = width
+        self.depth = depth
+        self.alpha = alpha
+        self.augmentations = augmentations
+
+    def apply_op(self, image, op):
+        image = op(image=image)['image']
+        return image
+
+    def apply(self, img, **params):
+        ws = np.float32(np.random.dirichlet([self.alpha] * self.width))
+        m = np.float32(np.random.beta(self.alpha, self.alpha))
+
+        flag_float = 0
+        if img.dtype in ['float64', 'float32']:
+            img_format = img.dtype
+            img = np.clip(img * 255., 0, 255).astype(np.uint8)
```

Comment:
why do we need this?

Reply:
some augmentations works only with uint8 images, with this row i'm avoiding exceptions with this type of augs

why do we need float format flag?
in particular case we use augs for float images and expect float images with return(actually, that's one of the test)
you can see usage of float flag and image format in return of function

Reply:
I think it's better to do the same thing as in other augmentations because silent casting and unnormalization could lead to tricky problems (there are cases when you float image is not normalized by only division by 255.

In this particular case I would leave image format as is and one of subsequent augmentations will throw exception.

Reply:
sounds reasonable